rskj-core/src/test/java/org/ethereum/vm/PrecompiledContractTest.java

```diff
@@ -228,4 +233,103 @@ public void modExpTest() {
+    }
+    @Test
+    public void blake2fTestVector6() {
+        String data ="0000000c48c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5d182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b61626300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000";
+        String expected = "75ab69d3190a562c51aef8d88f1c2775876944407270c42c9844252c26d2875298743e7f6d5ea2f2d3e8d226039cd31b4e426ac4f2d3d666a610c2116fde4735";
+
+        internalBlake2fTestVector(data, expected, null);
+    }
+    @Test
+    public void blake2fTestVector7() {
+        String data = "0000000148c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5d182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b61626300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000001";
+        String expected = "b63a380cb2897d521994a85234ee2c181b5f844d2c624c002677e9703449d2fba551b3a8333bcdf5f2f7e08993d53923de3d64fcc68c034e717b9293fed7a421";
+
+        internalBlake2fTestVector(data, expected, null);
+    }
+    @Test
+    public void blake2fTestVector8() {
+        // This tests takes a lot of time to be executed
+
+        String data = "ffffffff48c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5d182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b61626300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000001";
+        String expected = "fc59093aafa9ab43daae0e914c57635c5402d8e3d2130eb9b3cc181de7f0ecf9b22bf99a7815ce16419e200e01846e6b5df8cc7703041bbceb571de6631d2615";
+
+        internalBlake2fTestVector(data, expected, null);
+    }
+
+    private void internalBlake2fTestVector(String data, String expected, String expectedError) {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+
+        DataWord addr = DataWord.valueFromHex("0000000000000000000000000000000000000000000000000000000000000009");
+        PrecompiledContract contract = precompiledContracts.getContractForAddress(activations, addr);
+        assertNotNull(contract);
+
+        byte[] dataBytes = Hex.decode(data);
+
+        try {
+            byte[] resultBytes = contract.execute(dataBytes);
+
+            assertEquals(expected, Hex.toHexString(resultBytes));
```

Comment:
Be careful here.
I would leave the error assertion outside or in a new method, and add a "fail()" if it doesnt fail.
Could happen, that asserts / doesnt fail, and it will be still green when it shouldnt.

Reply:
modified!